### ISSUE TITLE
Some patternfly widgets are over-customized

### ISSUE THREAD

**Issue by dabeng** [patternfly/patternfly-3#605]
The over-customized widgets have serious influence for the popularity and development of patternfly. 

We know that patterns are the core section of patternfly. These patterns mostly come from the real applications or projects. While after we encapsulate a huge of features into the final widget, whether should we evaluate its reusability to check if it can be easily applied in the real projects. Complex demo is not equal to a reusable widget.

In short, what comes from the projects, goes to the projects. This is a virtuous cycle.
